include/mbedtls/x509_crt.h

```diff
@@ -501,6 +501,22 @@ int mbedtls_x509_crt_verify_restartable( mbedtls_x509_crt *crt,
                      void *p_vrfy,
                      mbedtls_x509_crt_restart_ctx *rs_ctx );
 
+/**
+ * \brief          Sort the server cert chain to handle potentially extraneous certificates and arbitrary orderings.
+ *                 For more detail pelease check https://tools.ietf.org/id/draft-ietf-tls-tls13-23.html#certificate.
+ *
+ * \note           From https://tools.ietf.org/id/draft-ietf-tls-tls13-23.html#certificate
+ *                 Prior to TLS 1.3, "certificate_list" ordering required each certificate to certify the one immediately
+ *                 preceding it; however, some implementations allowed some flexibility. Servers sometimes send both a
+ *                 current and deprecated intermediate for transitional purposes, and others are simply configured incorrectly,
+ *                 but these cases can nonetheless be validated properly. For maximum compatibility,
+ *                 all implementations SHOULD be prepared to handle potentially extraneous certificates and arbitrary orderings
+ *                 from any TLS version, with the exception of the end-entity certificate which MUST be first.
+ *
+ * \param chain    a certificate chain to be sorted
+ */
+void mbedtls_x509_crt_sort( mbedtls_x509_crt *chain );
```

Comment:
What's the rationale for making this part of the public API? I'd expect it to be an internal function only.

Reply:
This function is more like a utility for x509 certificate parsing, so I think put it in x509_crt.h/c and make it as a public API is straight forward, even only used by ssl_tls.

Reply:
@ArthurMa1978 We can always make a function public if we feel it's of general use, but we cannot easily undo it, so it needs to be carefully considered. In this case, I don't see a strong reason to introduce this function to the public API. Could you please make it internal?

Reply:
Make sense, fixed.